rts/motoko-rts-macros/src/lib.rs

```diff
@@ -167,3 +167,160 @@ pub fn non_incremental_gc(attr: TokenStream, input: TokenStream) -> TokenStream
+                    if attr.meta.to_token_stream().to_string() == "memory" {
+                        sort = ArgSort::Memory;
+                        false
+                    } else {
+                        true
+                    }
+                });
+                (
+                    sort,
+                    syn::Ident::new(&format!("arg{}", i), proc_macro2::Span::call_site()),
+                    (*pat.ty).clone(),
+                )
+            }
+        })
+        .collect();
+
+    // Motoko parameters
+    let params: Vec<proc_macro2::TokenStream> = fn_params
+        .iter()
+        .filter_map(|(sort, ident, _ty)| match sort {
+            ArgSort::Value => Some(quote!(#ident: crate::types::Value)),
+            ArgSort::Memory => None,
+        })
+        .collect();
+
+    // Memory parameter
+    let memory_ident = syn::Ident::new("memory", proc_macro2::Span::call_site());
+
+    // Motoko -> Rust conversions
+    let arg_conversions: Vec<proc_macro2::TokenStream> = fn_params
+        .iter()
+        .filter_map(|(sort, ident, _)| match sort {
+            ArgSort::Value => Some(quote!(
+                let #ident = crate::custom::FromValue::from_value(#ident, #memory_ident).unwrap()
+            )),
+            ArgSort::Memory => None,
+        })
+        .collect();
+
+    // Motoko args
+    let args: Vec<proc_macro2::TokenStream> = fn_params
+        .iter()
+        .map(|(sort, ident, _)| match sort {
+            ArgSort::Value => quote!(#ident),
+            ArgSort::Memory => quote!(#memory_ident),
+        })
+        .collect();
+
+    // Motoko return value
+    let ret = quote!(crate::types::Value);
+
+    // Wrapper function
+    let mut wrap_fn = fun.clone();
+    wrap_fn.sig.ident = syn::Ident::new(&format!("__motoko_{}", fn_ident), fn_ident.span());
+    let wrap_fn_ident = &wrap_fn.sig.ident;
+
+    // Custom section
+    let custom_section_ident = syn::Ident::new(
+        &format!("CUSTOM_SECTION_{}", fn_ident.to_string().to_uppercase()),
+        fn_ident.span(),
+    );
+    let custom_section_content = format!("{};", fn_name);
+    let custom_section_len = custom_section_content.len();
+    let custom_section_bytes =
+        syn::LitByteStr::new(custom_section_content.as_bytes(), fn_ident.span());
+
+    // Generated tokens
+    let output = quote!(
+        #wrap_fn
+
+        #[crate::ic_mem_fn(#ic_mem_attr)]
+        unsafe fn #fn_ident<M: crate::memory::Memory>(#memory_ident: &mut M, #(#params,)*) -> #ret {
+            #(#arg_conversions;)*
+            let ret = #wrap_fn_ident(#(#args,)*);
+            crate::custom::IntoValue::into_value(ret, #memory_ident).unwrap()
+        }
+
+        #[link_section = "rts:custom-functions"]
+        static #custom_section_ident: [u8; #custom_section_len] = *#custom_section_bytes;
+    );
+
+    // return syn::Error::new_spanned(quote!(), format!("Macro expansion:\n{}", output))
+    //     .to_compile_error()
+    //     .into();
```

Comment:
Cosmetics: Perhaps remove the commented code

Reply:
Replaced the commented code with conditional evaluation based on a `DEBUG` constant. There might be a better way to check the macro expansion, but this is the simplest solution that I've been able to find for gaining visibility into the generated code.